### ISSUE TITLE
Step Template with package reference and Version Control enabled, will attempt to resolve the package feed by using the "Feed ID" instead of "Feed Name"

### ISSUE THREAD

**Issue by danefalvo** [OctopusDeploy/Issues#7480]
### Team

- [X] I've assigned a team label to this issue

### Severity

Blocking all customers using CAC and a Step Template with a Package Reference

### Version

2022.1.2300

### Latest Version

I could reproduce the problem in the latest build

### What happened?

When you select "Create Release" on a project that contains a Step Template with a package reference, the page will stop loading after it attempts to identify the feed that is used by the Step Template. 

This is because the call it uses to match the feed looks like this:

`<apiendpoint>/skip=0&take=2147483647&name=<feed-id>`

when it should be 
`<apiendpoint>/skip=0&take=2147483647&name=<feed-name>`

Later versions of Octopus show a "Feed Error", while 2022.1.2300 doesn't.
![image](https://user-images.githubusercontent.com/5361797/162951505-7f72b571-43f6-4ac5-8f4b-d8dfb128f019.png)

### Reproduction

Reproduction available here:
https://octopus-operations.octopus.app/app#/Spaces-162/projects/feedid-not-feedname/deployments

### Error and Stacktrace

```shell
Request URL: https://octopus-operations.octopus.app/api/Spaces-162/feeds?skip=0&take=2147483647&name=Feeds-1261
Request Method: GET
Status Code: 200 
Remote Address: 52.250.93.139:443
Referrer Policy: no-referrer
```

### More Information

[Customer Ticket](https://octopus.zendesk.com/agent/tickets/87146)

### Workaround

The issue is with the API Call so although there is no workaround, you can still create a step (as opposed to relying on a step-template) and the package feed call works correctly.

**Comment by donnybell** [comment]
Looks similar to: https://github.com/OctopusDeploy/Issues/issues/7398

**Comment by YuKitsune** [comment]
I tried this on the latest version of 2022.1 and it seems to work.
Looks like it was fixed in 2022.1.2321 along with https://github.com/OctopusDeploy/Issues/issues/7462
